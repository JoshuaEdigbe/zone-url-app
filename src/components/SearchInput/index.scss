#search-input {
  width: 100%;
  box-sizing: border-box;
  height: 30px;
  border: 1px;
  background-color: #f3f3f3;
  padding: 5px 10px;
  outline: 0;
}
