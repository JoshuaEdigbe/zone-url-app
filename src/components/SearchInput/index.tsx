import React from "react";

import "./index.scss";

interface Props {
  onChange: any;
  value: string;
}

const SearchInput: React.FC<Props> = ({ onChange, value }) => {
  return (
    <div>
      <input type="text" onChange={onChange} id="search-input" value={value} placeholder="Search..." />
    </div>
  );
};

export default SearchInput;
