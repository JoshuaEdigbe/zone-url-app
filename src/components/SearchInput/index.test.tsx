import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import SearchInput from ".";

describe("SearchInput component", () => {
  const mockOnChange = jest.fn();

  it("renders without crashing", () => {
    render(<SearchInput onChange={mockOnChange} value="" />);
  });

  it("displays the correct initial value", () => {
    const { getByDisplayValue } = render(
      <SearchInput onChange={mockOnChange} value="initialValue" />
    );

    const inputElement = getByDisplayValue("initialValue");

    expect(inputElement).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <SearchInput onChange={mockOnChange} value="" />
    );
    expect(container).toMatchSnapshot();
  });
});
