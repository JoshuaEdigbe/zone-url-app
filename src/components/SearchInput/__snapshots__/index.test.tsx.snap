// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component matches snapshot 1`] = `
<div>
  <div>
    <input
      id="search-input"
      placeholder="Search..."
      type="text"
      value=""
    />
  </div>
</div>
`;
